File: Intro/apple-division.cpp
#include <iostream>
#include <vector>
#include <cmath>

int main() {
    int n;
    std::cin >> n;

    std::vector<int> arr(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> arr[i];
    }

    long long mn = 1e18;

    for (int i = 0; i < (1 << n); ++i) {
        long long sm1 = 0, sm2 = 0;
        for (int j = 0; j < n; ++j) {
            if ((1 << j) & i) {
                sm1 += arr[j];
            } else {
                sm2 += arr[j];
            }
        }

        mn = std::min(mn, std::abs(sm1 - sm2));
    }

    std::cout << mn << std::endl;

    return 0;
}

File: Intro/apple-division.py
n = int(input())

arr = [int(x) for x in input().split()]

mn = 10 ** 9

for i in range(1<<n):
    sm1, sm2 = 0, 0
    for j in range(n):
        if 1<<j & i >0:
            sm1 += arr[j]
        else:
            sm2 += arr[j]

    mn = min(mn, abs(sm1-sm2))

print(mn)